src/core/server/plugins/plugin_context.ts

```diff
@@ -109,6 +110,8 @@ export function createPluginSetupContext<TPlugin, TPluginDependencies>(
     },
     http: {
       createCookieSessionStorageFactory: deps.http.createCookieSessionStorageFactory,
+      createRouter: () => deps.http.createRouter(`/${plugin.name}`),
+      registerRouter: (router: IRouter) => deps.http.registerRouter(router),
```

Comment:
From a plugin's perspective, it seems a bit strange you have to call `createRouter` and `registerRouter`. It seems like createRouter could be changed to and no `registerRouter` would need to be exposed:
```js
() => {
  const newRouter = deps.http.createRouter(`/${plugin.name}`);
  deps.http.registerRouter(router);
  return newRouter;
}
```

Reply:
we can reduce api surface for core services as well, will move `registerRouter` under `createRouter` within HttpService